changesets2CSV/changesets2CSV

```diff
@@ -121,6 +173,308 @@ def count_new_modified_deleted(changeset):
     newModifiedDeleted['Deleted'] = len(root.findall('delete'))
     return newModifiedDeleted
 
+def overpass_status(api_status_url = "https://overpass-api.de/api/status"):
+    """Get the overpass status -- this returns an int with the time to wait"""
+    session = requests.session()
+    session.headers.update({'User-Agent': OVERPASS_USER_AGENT})
+    cached_session = CacheControl(session)
+    response = cached_session.get(api_status_url)
+    if (response.status_code != requests.codes.ok):
+        raise ValueError("Bad Request: {}".format(api_status_url))
+    parsed_response = {'wait_time': []}
+    for i in response.text.splitlines():
+        if "Connected as" in i:
+            parsed_response['connected_as'] = i.split(":")[1].strip()
+        elif "Current time" in i:
+            parsed_response['current_time'] = i.split(":")[1].strip()
+        elif "Rate limit" in i:
+            parsed_response['rate_limit'] = int(i.split(":")[1].strip())
+        elif "slots available now" in i:
+            parsed_response['slots_available'] = int(i.split(" ")[0].strip())
+        elif "Slot available after" in i:
+            parsed_response['wait_time'].append(int(i.split(" ")[5]))
+            print('Adding wait_time of ', parsed_response['wait_time'])
+    if 'slots_available' not in parsed_response:
+        parsed_response['slots_available'] = 0
+    wait_time = 0
+    if parsed_response['rate_limit'] - parsed_response['slots_available'] >= 2 and len(parsed_response['wait_time']) > 0:
+        #We use max to ensure that there is always an open slot
+        return max(parsed_response['wait_time'])
+    return wait_time
+
+def overpass_query(query):
+    """Query the overpass servers. This may block for extended periods of time, depending upon the query"""
+
+    session = requests.session()
+    session.headers.update({'User-Agent': OVERPASS_USER_AGENT})
+    cached_session = CacheControl(session)
+    response = cached_session.post("http://overpass-api.de/api/interpreter", data={'data': query})
```

Comment:
> let's reformat this a little (use tabs and such instead of spaces):
> 
> ```
> response = cached_session.post(
>                        "http://overpass-api.de/api/interpreter",
>                        data={'data': query}
>                     )
> ```

Will do